0027-safely-recommend-actions.adoc

```diff
@@ -0,0 +1,173 @@
+
+== Swap State Machine
+
+A successful COMIT atomic swap (happy path) may be expressed as a series of state transitions:
+
+        Created --> Alpha Funded --> Beta Funded --> Beta Redeemed --> Alpha Redeemed --> Done
+
+Each transition is associated with a specific action named accordingly e.g., 'fund alpha' transitions from created to alpha funded.
+
+For each state transition there is a time requirement, henceforth called the transition period. The total time requirement for a swap is the sum of the transition periods. A transition period is effected by the state we are transitioning from and the state we are transitioning to, each transition period therefore is calculated differently.
+
+== Transition Period
+
+There exist a number of factors, that when combined, represent the transition period.
+
+.Considerations
+* Time to Act (`TTA`)
+* Time to Inclusion (`TTI`)
+* Time to Final (`TTF`)
+* HTLC Expiry time
+
+=== Time to Act
+
+For each actor there is a time frame within which they can be expected to act.
+
+Different users will have different response times e.g., a long running daemon is likely to respond to swap transitions more quickly than a user who is manually checking an application before doing each action. We define Time To Act (`TTA`) as the window of time within which the counterparty can expected to take the next action.
+
+=== Time To Inclusion
+
+For each blockchain there exists a time for a transaction to be included in a block.
+
+The likelihood of a transaction being included in a block is dependant on the transaction fee coupled with the current load on the blockchain network i.e., the number of current unconfirmed transactions. An increase in the transaction fee leads to an increase in the probability that the transaction will be included. Inclusion is not guaranteed, therefore the probability is defined to be the percentage chance of a transaction being included in a block within the next `N` blocks.
+
+With a given fee and a given probability tolerance we define Time To Inclusion for a given blockchain as:
+
+----
+TTI = N * <average block time>
+----
+
+=== Time To Final
+
+For each blockchain there exists a time, after a transaction has been included in a block, that the transaction is considered to be final.
+
+Confirmations represent the number of blocks in the block chain that have been accepted by the network since the block that includes the transaction. The number of confirmations a transaction has determines the probability that a blockchain fork will invalidate the transaction. With a given blockchain we define Time To Final as:
```

Comment:
```suggestion
Confirmations represent the number of blocks in the blockchain that have been accepted by the network since the block that includes the transaction. The number of confirmations a transaction has determines the probability that a blockchain fork will invalidate the transaction. With a given blockchain we define Time To Final as:
```

Reply:
Fixed, thanks.